examples/tutorial/TUTORIAL.md

```diff
@@ -0,0 +1,191 @@
+                 Right (Just (OneInt x)) -> loop (i + x)
+                 Right Nothing -> return (ServerReaderResponse
+                                           (Just (OneInt i))
+                                           []
+                                           StatusOk
+                                           "")
+```
+
+The `ServerReaderResponse` type is almost the same as `ServerNormalResponse`, except that the first argument, the message to send back to the client, is optional. Otherwise, it takes metadata (which we leave empty), a status code, and a string containing more information about the status code.
+
+### The client
+
+The client-side code generated for us is `arithmeticClient`, which takes a `Client` as input and gives us a record containing actions that execute RPCs. To start up the C gRPC library and get a `Client`, we use `withGRPCClient`, which takes a `ClientConfig`:
+
+
+
+```
+clientConfig :: ClientConfig
+clientConfig = ClientConfig { clientServerHost = "localhost"
+                            , clientServerPort = 50051
+                            , clientArgs = []
+                            , clientSSLConfig = Nothing
+                            }
+
+main :: IO ()
+main = withGRPCClient clientConfig $ \client -> do
+  (Arithmetic arithmeticAdd arithmeticRunningSum) <- arithmeticClient client
+  -- to be continued!
+```
+
+Now that we are on the client side, the `Arithmetic` record contains functions that make RPC requests. You can think of these functions as roughly having the type `ClientRequest -> ClientResult`. Like before, the particular constructors will vary depending on whether the RPC is streaming or not.
+
+#### Requesting unary RPC
+
+Here we construct a `ClientNormalRequest`, which takes as input a message, a timeout in seconds, and metadata. The result is a `ClientNormalResponse`, containing our the server's response, the initial and trailing metadata for the call, and the status and status details string.
+
+```
+-- Request for the Add RPC
+  ClientNormalResponse (OneInt x) _meta1 _meta2 _status _details
+    <- arithmeticAdd (ClientNormalRequest (TwoInts 2 2) 1 [])
+  print ("2 + 2 = " ++ (show x))
+```
+
+#### Executing a client streaming RPC
+
+Doing a streaming request is slightly trickier. As input to the streaming RPC action, we pass in another IO action that tells `grpc-haskell` what to send. It takes a `send` action as input. This is a bit convoluted, but it guarantees that you can't send streaming messages outside of the context of a streaming call!
+
+```
+-- Request for the RunningSum RPC
+ClientWriterResponse reply _streamMeta1 _streamMeta2 streamStatus streamDtls
+  <- arithmeticRunningSum $ ClientWriterRequest 1 [] $ \send -> do
+      _ <- send (OneInt 1)
+      _ <- send (OneInt 2)
+      _ <- send (OneInt 3)
+      return ()
+```
+
+Each `send` potentially returns an error message, but for the purposes of this tutorial, we skip the error checking and throw away the result.
```

Comment:
As far as I know, there are no recoverable errors that can happen when streaming. Which is probably worth mentioning here. I'll change it to print the error, at least.